File: SharedObjects/SharedObjects/Models/TaskHistory.cs
using System.ComponentModel.DataAnnotations;

namespace SharedObjects.Models;

public class TaskHistory
{
    [Key] public Guid Id { get; set; }

    public Guid TaskId { get; set; }
    public Task Task { get; set; } = null!;

    public DateTime ChangeDate { get; set; }

    public int OldStatusId { get; set; }
    public TaskStatus OldStatus { get; set; } = null!;

    public int NewStatusId { get; set; }
    public TaskStatus NewStatus { get; set; } = null!;
}

File: laravel/app/Models/TaskHistory.php
<?php

/**
 * Created by Reliese Model.
 */

namespace App\Models;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Model;

/**
 * Class TaskHistory
 * 
 * @property uuid $Id
 * @property uuid $TaskId
 * @property Carbon $ChangeDate
 * @property string $NewStatus
 * @property string|null $OldStatus
 * 
 * @property Task $task
 *
 * @package App\Models
 */
class TaskHistory extends Model
{
	protected $table = 'TaskHistories';
	protected $primaryKey = 'Id';
	public $incrementing = false;
	public $timestamps = false;

	protected $casts = [
		'Id' => 'uuid',
		'TaskId' => 'uuid',
		'ChangeDate' => 'datetime'
	];

	protected $fillable = [
		'TaskId',
		'ChangeDate',
		'NewStatus',
		'OldStatus'
	];

	public function task()
	{
		return $this->belongsTo(Task::class, 'TaskId');
	}
}
